### ISSUE TITLE
[Android] Use more "Others" memory than the native demo

### ISSUE THREAD

**Issue by littleGnAl** [flutter/flutter#140065]
### Steps to reproduce

I am working on an optimization of our video rendering widget. After profiling, I found that there're the extra memory comes from the "Others" memory.

Native demo
![image](https://github.com/flutter/flutter/assets/8847263/22945947-4faf-490e-8732-dc29a86b794a)

Flutter demo
![image](https://github.com/flutter/flutter/assets/8847263/a4a75f84-e3b2-4196-828d-5b14e36c95d6)

And although just for a hello world example, the "Others" memory is higher than the native demo.
<img width="773" alt="image" src="https://github.com/flutter/flutter/assets/8847263/916c2d98-b62e-46fd-8415-1a0b8ccc4673">

After some search, I found this issue https://github.com/flutter/flutter/issues/48391 for reference, and try to set the `Skia.setResourceCacheMaxBytes` to a lower value to see if the "Others" memory is raised by the skia cache,
```dart
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await SystemChannels.skia.invokeMethod('Skia.setResourceCacheMaxBytes', 1024 * 1024 * 5);
  runApp(const MyApp());
}
```
But it seems nothing to work. I want to figure out why it uses more "Others" memory than the native demo, and is there a way to reduce "Others" memory?

### Code sample

I believe this issue can be reproduced by a new example.
Run `flutter create hello`, and then profile the memory usage.

### What target platforms are you seeing this bug on?

Android

### OS/Browser name and version | Device information

Device: Oppo k5
System version: Android 11
Display: 1080x2340

### Does the problem occur on emulator/simulator as well as on physical devices?

Unknown

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.9, on macOS 14.0 23A344 darwin-arm64, locale zh-Hans-CN)
    • Flutter version 3.13.9 on channel stable at /Users/admin/Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (7 weeks ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/admin/Library/Android/sdk
    • Platform android-34, build-tools 33.0.1
    • ANDROID_HOME = /Users/admin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.82.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • PCNM00 (mobile) • b97fe38 • android-arm64  • Android 11 (API 30)
    • macOS (desktop) • macos   • darwin-arm64   • macOS 14.0 23A344 darwin-arm64
    • Chrome (web)    • chrome  • web-javascript • Google Chrome 120.0.6099.71

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by huycozy** [comment]
I checked this with two newly created Flutter and Android native projects and used Android Studio's Profiler. 

For Flutter, run project in `profile` mode. For Android, run project with "Profile `app` with complete data" option from Android Studio.

Although `Others` memory is not as different as the original report (10x), I still can see a difference between Flutter and the native app here:

- Flutter Android app `Others` memory: 19.2MB
- Android native app `Others` memory: 9.8MB

### Flutter
![Screenshot 2023-12-14 at 14 47 50](https://github.com/flutter/flutter/assets/104349824/7d844963-dd4a-4841-bbeb-63b6afe4809e)

### Android
![Screenshot 2023-12-14 at 14 55 44](https://github.com/flutter/flutter/assets/104349824/cfd05571-fe74-4e49-a0b8-3735b6a16146)

Labeling the issue for other's insights. 

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.4, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.4 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2e9cb0aa71 (2 days ago), 2023-12-11 14:35:13 -0700
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.85.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 120.0.6099.71

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.18.0-11.0.pre.32, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.18.0-11.0.pre.32 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3b370c8539 (2 hours ago), 2023-12-14 01:55:57 +0100
    • Engine revision 45b95f264d
    • Dart version 3.3.0 (build 3.3.0-219.0.dev)
    • DevTools version 2.30.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.85.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.78.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 120.0.6099.71

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>

**Comment by camsim99** [comment]
Re-routing to the engine team for further insight, but feel free to hand back to the Android team.

**Comment by littleGnAl** [comment]
I'm so sorry that I made a big mistake when I opened the Flutter project as an Android project (via opening the `myproject/android/setting.gradle`), in this way, when I click the profile button the Flutter profile configuration does not take the effect, and it built as a debug apk. After correcting my operation to open the Flutter project in Android Studio, and profile it, the "Others" memory performance is similar to the results of @huycozy  test. I will update my issue. 

native demo:
![image](https://github.com/flutter/flutter/assets/8847263/c622b8a2-88d2-4749-979a-d339f542856f)

flutter demo
![image](https://github.com/flutter/flutter/assets/8847263/22436c54-3337-48c6-83db-21b139b1c007)


But I still want to figure it out why there're more than about 10MB memories than the native demo, and is there way to reduce it.
